### PR TITLE
Article on What is rest API

### LINKED ISSUE #359: New article: What is Rest API
# Title: [Article] What is Rest API

- [X ] What is this about: What is a good API and how should I create one? Which API practices are important so my API will have good DX?  

- [x ] Read more here: The starting point for your research (You can use different sources if you have better ones) [https://apifriends.com/api-management/what-is-an-api/](https://apifriends.com/api-management/what-is-an-api/), [https://medium.com/@mwaysolutions/10-best-practices-for-better-restful-api-cbe81b06f291](https://medium.com/@mwaysolutions/10-best-practices-for-better-restful-api-cbe81b06f291), [https://docs.microsoft.com/cs-cz/azure/architecture/best-practices/api-design](https://docs.microsoft.com/cs-cz/azure/architecture/best-practices/api-design), [https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/](https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/), [https://florimond.dev/blog/articles/2018/08/restful-api-design-13-best-practices-to-make-your-users-happy/](https://florimond.dev/blog/articles/2018/08/restful-api-design-13-best-practices-to-make-your-users-happy/), [https://swagger.io/resources/articles/best-practices-in-api-design/](https://swagger.io/resources/articles/best-practices-in-api-design/), [https://stackoverflow.com/questions/21660791/what-is-the-main-difference-between-patch-and-put-request?answertab=oldest#tab-top](https://stackoverflow.com/questions/21660791/what-is-the-main-difference-between-patch-and-put-request?answertab=oldest#tab-top).

- [x ] Pick the corresponding labels: 
    - Stage:development

## How to contribute:
- Follow our [Contribution guide](https://github.com/DXHeroes/knowledge-base-content/blob/master/CONTRIBUTING.md)
- How to [Git and Pull Requests]( https://github.com/firstcontributions/first-contributions/blob/master/README.md)
- How to [Markdown](https://guides.github.com/features/mastering-markdown/), [Markdown interactive demo](https://www.markdowntutorial.com/lesson/1/) or [Google Docs Addon](https://gsuite.google.com/marketplace/app/docs_to_markdown/700168918607)

### PR DESCRIPTION
**Pull request by D-stevpm**
Closes #359 

## If an article related PR

- [ ] I have checked that my suggestion follows the Contribution guide

- [ ] I have updated all sections as requested 

- [ ]  I have added which issue this PR closes

- [ ] I have 0 warnings from Danger.js

- [ ] If I have added images, they are in /files/ folder

## If other PR

- [ ] I have added which issue this PR closes

### CODE REVIEW

File: practices/what_is_rest_api [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+As a popular architectural style for designing and developing web-based APIs. It helps to solve a variety of problems related to building web-based applications and services, including:
+
+*  **Scalability**: RESTful APIs are highly scalable due to their stateless nature, which allows for easy scaling of resources.
+*  **Flexibility**: RESTful APIs are flexible in terms of the format of data, which allows for easy integration with other web-based systems.
+*  **Easy to understand**: RESTful APIs are easy to understand due to their simplicity and uniform interface.
+*  **Easy to develop**: RESTful APIs are easy to develop due to their uniform interface and stateless nature, which reduces the complexity of the application.
+*  **Platform independence**: RESTful APIs can be used with any programming language and platform, making them highly interoperable.
+
+### How to Implement RESTful API
```
Comment (prokopsimek) [inline practices/what_is_rest_api:71]:
```suggestion
## How to Implement RESTful API
```

**Comment by D-stevpm** [comment]
@prokopsimek Thanks for the merge, Please what is next? Can I suggest new issues or what do you suggest

**Comment by prokopsimek** [comment]
@D-stevpm Take a look at your amazing article https://developerexperience.io/articles/rest-api first! 👏

Now that the merge is complete, it's time to move on to the next tasks. If you have any new issues or ideas, please feel free to suggest them. It would be great to hear your thoughts on what we can work on next. Additionally, if you have any questions or need any guidance, don't hesitate to ask. Let's keep up the momentum!

**Comment by D-stevpm** [comment]
It is amazing, Thank you. Is it possible to get analytics on how the content performed in the next month? @prokopsimek 

Time to move to the next. 
Cheers

**Comment by prokopsimek** [comment]
@D-stevpm Sure, I can definitely send you some analytics after a month, but keep in mind that the impact of the content may not be fully reflected in the analytics in the first month due to the time it takes for Google and other search engines to index the content. Therefore, it may be best to wait until the third month to get a more accurate view of the content's performance.

**Comment by D-stevpm** [comment]
All right, Thank you

**Pull request merged**